Tasks/AzureNLBManagement/nlbTask.ts

```diff
@@ -0,0 +1,102 @@
+		var action: string = tl.getInput("Action", true);
+		var nicDetection: string = tl.getInput("NICDetection", true);		
+		var vmNicMapping: string = tl.getInput("VmNicMapping", false);
+		var endPointAuthCreds = tl.getEndpointAuthorization(connectedServiceName, true);
+		var endpointUrl = tl.getEndpointUrl(connectedServiceName, true);
+
+		var SPN = new Array();
+		SPN["servicePrincipalClientID"] = endPointAuthCreds.parameters["serviceprincipalid"];
+		SPN["servicePrincipalKey"] = endPointAuthCreds.parameters["serviceprincipalkey"];
+		SPN["tenantID"] = endPointAuthCreds.parameters["tenantid"];
+		SPN["subscriptionId"] = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true);
+
+		var nicVm = await getNetworkInterface(SPN, endpointUrl, resourceGroupName, nicDetection, vmNicMapping);
+		tl.debug(tl.loc("NicDetailsFetched", process.env["computername"]));
+
+		var nicLbBackendPoolConfig = null;
+		if (action == "Connect") {
+			tl._writeLine(tl.loc("ConnectingVMtoLB", loadBalancerName));
+			var lb = await nlbUtility.getLoadBalancer(SPN, endpointUrl, loadBalancerName, resourceGroupName);
+			nicLbBackendPoolConfig = lb.properties.backendAddressPools;
+		}
+		else {
+			tl._writeLine(tl.loc("DisconnectingVMfromLB", loadBalancerName));
+		}
+		nicVm.properties.ipConfigurations[0].properties['loadBalancerBackendAddressPools'] = nicLbBackendPoolConfig;
+		var setNIStatus = await nlbUtility.setNetworkInterface(SPN, endpointUrl, nicVm, resourceGroupName);
+		tl._writeLine(tl.loc(setNIStatus, nicVm.name));
+		tl._writeLine(tl.loc("ActionCompletedSuccefully", action, process.env.computername, loadBalancerName));
+	}
+	catch(error) {
+		tl.setResult(tl.TaskResult.Failed, error);
+	}
+}
+
+function getMacAddress(): string[] {
+	// Return an array of mac address of all the network interfaces 
+	var macAddress = [];
+	var networkInterfaces = os.networkInterfaces();
+	Object.keys(networkInterfaces).forEach( (interfaceName) => {
+		networkInterfaces[interfaceName].forEach( (interFace) => {
+			if (interFace.family !== 'IPv4' || interFace.internal !== false) {
+      			return;
+    		}
+    		macAddress.push(interFace.mac.toUpperCase().replace(/:/g, "-"));
+		});
+	});
+	return macAddress;
+}
+
+async function getNetworkInterface(SPN, endpointUrl: string, resourceGroupName: string, nicDetection: string, vmNicMapping: string) {
+	var nics =  await nlbUtility.getNetworkInterfacesInRG(SPN, endpointUrl, resourceGroupName);
+	var macAddress = getMacAddress();
+	var nicVm = null;
+	if(nicDetection == "AutoDetectNic"){
+		tl.debug(tl.loc("GettingPrimaryNicForVm", process.env.computername));
+		for (var mac in macAddress) {
+			for (var i in nics) {
+				if(nics[i].properties.macAddress == macAddress[mac] && nics[i].properties.primary) {
+					nicVm = nics[i];
+					break;
+				}
+			}	
+		}
+	}
+	else {
+		// Handle custom vm:nic mapping 
+		var nicVmMap = {};	
+		var temp = vmNicMapping.split(" ");
```

Comment:
You have to validate that vmNicMapping input is in right format. Any improper input with wrong : or wrong spaces can lead to failure below while you use the split out as index to arrays.